### ISSUE TITLE
create_sql_agent Error: table_names {"'<table name>'"} not found in database.

### ISSUE THREAD

**Issue by yernenip** [langchain-ai/langchain#12911]
### System Info

LangChain version: 0.0.330
Windows 11
Python 3.11.3
SQLAlchemy version: 2.0.23

### Information

- [ ] The official example notebooks/scripts
- [X] My own modified scripts

### Related Components

- [ ] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [ ] Vector Stores / Retrievers
- [ ] Memory
- [X] Agents / Agent Executors
- [X] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

Steps to reproduce the behavior.

1. Follow the steps mentioned in https://python.langchain.com/docs/integrations/toolkits/sql_database
2.  Replace connection string with a PostgreSQL connection string
3.  Run the following code, make sure to update the connection string and the database/tables you are querying.

import langchain
from langchain.llms import CTransformers
from langchain.agents import create_sql_agent
from langchain.agents.agent_toolkits import SQLDatabaseToolkit
from langchain.sql_database import SQLDatabase
from langchain.agents import AgentExecutor
from langchain.agents.agent_types import AgentType

config = {'max_new_tokens': 256, 'repetition_penalty': 1.1, 'temperature': 0, 'context_length': 4096}

llm = CTransformers(model="TheBloke/CodeLlama-7B-Instruct-GGUF", 
                    model_file="codellama-7b-instruct.Q4_K_M.gguf",config=config, verbose=True)

CONNECTION_STRING = 'postgresql://postgresuser:password@localhost:5432/pharmadb'
db = SQLDatabase.from_uri(CONNECTION_STRING)

toolkit = SQLDatabaseToolkit(db=db, llm=llm)

agent_executor = create_sql_agent(
    llm=llm,
    toolkit=toolkit,
    verbose=True,
    agent_type=AgentType.ZERO_SHOT_REACT_DESCRIPTION,
)

agent_executor.run("Describe the corporation table")

I get the following output in verbose mode..

> Entering new AgentExecutor chain...
Action: sql_db_list_tables
Action Input: ''
Observation: corporation, molecule, product_mat, productpack
Thought: The corporation table is probably one of those tables.
Action: sql_db_schema
Action Input: 'corporation'
Observation: Error: table_names {"'corporation'"} not found in database
Thought: I must have misspelled the name of the table.  Let me try again.
Action: sql_db_schema
Action Input: 'corporation'
Observation: Error: table_names {"'corporation'"} not found in database
Thought: I must have misspelled the name of the table.  Let me try again.
Action: sql_db_schema
Action Input: 'corporation'
Observation: Error: table_names {"'corporation'"} not found in database
Thought: I must have misspelled the name of the table.  Let me try again.
Action: sql_db_schema
Action Input: 'corporation'
Observation: Error: table_names {"'corporation'"} not found in database
Thought: I must have misspelled the name of the table.  Let me try again.
Action: sql_db_schema
Action Input: 'corporation'
Observation: Error: table_names {"'corporation'"} not found in database
Thought: I must have misspelled the name of the table.  Let me try again.
Action: sql_db_schema
Action Input: 'corporation'
.
.
.

> Finished chain.
'Agent stopped due to iteration limit or time limit.'

### Expected behavior

Expected to correctly identify the table and its schema, run the query to retrieve top 3 records from the table and provide a description of the table.

**Comment by fr3fou** [comment]
Did you ever figure out a solution? @yernenip

**Comment by yernenip** [comment]
@fr3fou Not really, I moved on from LangChain to using function calling from OpenAI

**Comment by dijkstra-mose** [comment]
I also encountered the same problem. After I upgrade the langchain, the problem is fixed. I think your langchain version is too early.
